crates/sui-graphql-rpc/src/server/version.rs

```diff
@@ -132,6 +112,10 @@ fn parse_version(version: &str) -> Option<(&str, &str)> {
     let year = parts.next()?;
     let month = parts.next()?;
 
+    if year.is_empty() || month.is_empty() {
+        return None;
+    }
+
```

Comment:
This checks for this case when `2024.` is passed
```
year = "2024"
month = ""
```
which surprisingly, the `month.chars().all(....)` returns true, instead of false.